<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Dilara Ayyildiz" />

<meta name="date" content="2024-07-19" />

<title>Figure1_plots</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Adult_hippocampal_neurogenesis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Figure1_plots</h1>
<h4 class="author">Dilara Ayyildiz</h4>
<h4 class="date">2024-07-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-04-17
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>Adult_hippocampal_neurogenesis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250417code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250417)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250417code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250417)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong8be8d07">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 8be8d07
</a>
</p>
</div>
<div id="strongRepositoryversionstrong8be8d07"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 8be8d07.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/figure/
    Ignored:    data/.DS_Store
    Ignored:    data/CellPhone_sigInteractions.txt
    Ignored:    data/Donor_info_heatmap.xlsx
    Ignored:    data/Fig4D_Adult_CTR_ImNs_vs_Fetal_ImNB_rldmatrix.txt
    Ignored:    data/Fig4E_ImNs_vs_MatNs_vsdmatrix.txt
    Ignored:    data/Group_matrices/
    Ignored:    data/RNAscope_combo3.txt
    Ignored:    data/Summary_spheroids_D0-D16-D30-D50-D75_qPCR-calculations.xlsx
    Ignored:    data/adult_QC_thresholds.txt
    Ignored:    data/adult_filtered.RDS
    Ignored:    data/adult_metadata_info.txt
    Ignored:    data/adult_raw.RDS
    Ignored:    data/adultgc_emptydroplets_integrated.RDS
    Ignored:    data/adultgc_fetalNeurolin_int.RDS
    Ignored:    data/adultgc_fetalNeurolin_int_subset.RDS
    Ignored:    data/adultgc_filtered.RDS
    Ignored:    data/fetal_QC_thresholds.txt
    Ignored:    data/fetal_filtered.RDS
    Ignored:    data/fetal_raw.RDS
    Ignored:    data/libd_stemcell_timecourse_rseGene_n157.rda
    Ignored:    data/marker_genesets.xlsx
    Ignored:    data/neurolin.RDS
    Ignored:    data/neurolin_sce.RDS
    Ignored:    data/neurolin_sce_fitgam.RDS
    Ignored:    data/overlaping_interactions_CellPhone_CellChat.txt
    Ignored:    data/rlog_corrected_ImNs_SAD_vs_RES.RDS
    Ignored:    data/tradeseq_neurolin_heatmapdata.txt
    Ignored:    output/.DS_Store
    Ignored:    output/Figures/
    Ignored:    output/SupFigures/
    Ignored:    output/SupTables/

Untracked files:
    Untracked:  analysis/Fig1_script.Rmd
    Untracked:  analysis/Functions.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="qc-processing" class="section level1">
<h1>QC processing</h1>
<pre class="r"><code># read raw file
library(here)</code></pre>
<pre><code>here() starts at /Users/dilaraayyildiz/saltalab_github_repos/Adult_hippocampal_neurogenesis</code></pre>
<pre class="r"><code>source(here(&quot;analysis&quot;, &quot;Functions.R&quot;))</code></pre>
<pre><code>Warning: package &#39;ggplot2&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;cli&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;clusterProfiler&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;S4Vectors&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;GenomeInfoDb&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;dunn.test&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;edgeR&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;Rcpp&#39; was built under R version 4.3.3</code></pre>
<pre><code>Warning: package &#39;AnnotationDbi&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning: package &#39;UCell&#39; was built under R version 4.3.2</code></pre>
<pre><code>Warning in RNGkind(&quot;Mersenne-Twister&quot;, &quot;Inversion&quot;, &quot;Rounding&quot;): non-uniform
&#39;Rounding&#39; sampler used</code></pre>
<pre class="r"><code>adult&lt;-readRDS(here(&quot;data&quot;,&quot;adult_filtered.RDS&quot;))</code></pre>
<div id="supp-fig1a---pre-integration" class="section level3">
<h3>Supp Fig1A - pre integration</h3>
<pre class="r"><code># tiff(here(&quot;output&quot;, &quot;SupFigures&quot;,&quot;SupFig1A_1.tiff&quot;), units=&quot;in&quot;, width=8, height=6, res=300)
DimPlot(adult, reduction = &quot;umap.unintegrated&quot;, group.by = &quot;group&quot;, label = F, cols = CFG$groupspal, pt.size = 0.01, raster=FALSE) + xlab(&quot;umap_1&quot;) + ylab(&quot;umap_2&quot;)+theme(axis.text.x=element_text(size=20),axis.text.y=element_text(size=20), axis.title.x = element_text(size=20), axis.title.y = element_text(size=20), legend.text = element_text(size=20), title = element_text(size=20))</code></pre>
<p><img src="figure/Fig1_script.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dev.off()

# tiff(here(&quot;output&quot;, &quot;SupFigures&quot;,&quot;SupFig1A_2.tiff&quot;), units=&quot;in&quot;, width=8, height=6, res=300)
DimPlot(adult, reduction = &quot;umap.unintegrated&quot;, group.by = &quot;Run&quot;, label = F, cols = CFG$mypal[11:16], pt.size = 0.01, raster=FALSE) + xlab(&quot;umap_1&quot;) + ylab(&quot;umap_2&quot;)+theme(axis.text.x=element_text(size=20),axis.text.y=element_text(size=20), axis.title.x = element_text(size=20), axis.title.y = element_text(size=20), legend.text = element_text(size=20), title = element_text(size=20))</code></pre>
<p><img src="figure/Fig1_script.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dev.off()

# tiff(here(&quot;output&quot;, &quot;SupFigures&quot;,&quot;SupFig1A_3.tiff&quot;), units=&quot;in&quot;, width=8, height=6, res=300)
DimPlot(adult, reduction = &quot;umap.unintegrated&quot;, group.by = &quot;sample&quot;, label = F, cols = CFG$mypal, pt.size = 0.01, raster=FALSE) + xlab(&quot;umap_1&quot;) + ylab(&quot;umap_2&quot;)+theme(axis.text.x=element_text(size=20),axis.text.y=element_text(size=20), axis.title.x = element_text(size=20), axis.title.y = element_text(size=20), legend.text = element_text(size=20), title = element_text(size=20))</code></pre>
<p><img src="figure/Fig1_script.Rmd/unnamed-chunk-2-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dev.off()</code></pre>
</div>
<div id="supp-fig1b" class="section level3">
<h3>Supp Fig1B</h3>
<pre class="r"><code>adult_raw&lt;-readRDS(here(&quot;data&quot;,&quot;adult_raw.RDS&quot;))
qc_thresholds &lt;- read.table(here(&quot;data&quot;,&quot;adult_QC_thresholds.txt&quot;), header = T, sep=&#39;\t&#39;)

samples&lt;-unique(adult_raw$sample)
feature_plots &lt;- map(samples, function(sample_id) {
  print(paste(&quot;Processing sample:&quot;, sample_id))
  sObj &lt;- subset(adult_raw, subset = sample == sample_id)
  fun_nFeatureRNA_plot(sObj, sample_id, qc_thresholds)
})</code></pre>
<pre><code>[1] &quot;Processing sample: S1R1&quot;
[1] &quot;Processing sample: S1R2&quot;
[1] &quot;Processing sample: S2R2&quot;
[1] &quot;Processing sample: S3R2&quot;
[1] &quot;Processing sample: S4R2&quot;
[1] &quot;Processing sample: S2R3&quot;
[1] &quot;Processing sample: S1R3&quot;
[1] &quot;Processing sample: S1R4&quot;
[1] &quot;Processing sample: S2R4&quot;
[1] &quot;Processing sample: S3R4&quot;
[1] &quot;Processing sample: S4R4&quot;
[1] &quot;Processing sample: S1R5&quot;
[1] &quot;Processing sample: S2R5&quot;
[1] &quot;Processing sample: S3R5&quot;
[1] &quot;Processing sample: S4R5&quot;
[1] &quot;Processing sample: S5R5&quot;
[1] &quot;Processing sample: S6R5&quot;
[1] &quot;Processing sample: S7R5&quot;
[1] &quot;Processing sample: S8R5&quot;
[1] &quot;Processing sample: S1R5stored&quot;
[1] &quot;Processing sample: S2R5stored&quot;
[1] &quot;Processing sample: S1R6&quot;
[1] &quot;Processing sample: S2R6&quot;
[1] &quot;Processing sample: S5R6&quot;</code></pre>
<pre class="r"><code># pdf(here(&quot;output&quot;, &quot;SupFigures&quot;,&quot;SupFig1B_1.pdf&quot;), width=20, height=16)
do.call(grid.arrange, c(feature_plots, nrow = 4, ncol = 6))</code></pre>
<p><img src="figure/Fig1_script.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dev.off()

metadata &lt;- adult_raw@meta.data

# pdf(here(&quot;output&quot;, &quot;SupFigures&quot;,&quot;SupFig1B_2.pdf&quot;), width=14, height=8)
metadata$percent_ribo[is.nan(metadata$percent_ribo)] &lt;- 0  # Replace NaN with 0
metadata$percent_ribo[is.na(metadata$percent_ribo)] &lt;- 0   # Replace NA with 0
metadata$percent_ribo[metadata$percent_ribo == 0] &lt;- 0.1  # avoid inf values
medians &lt;- metadata %&gt;%
  group_by(sample) %&gt;%
  summarize(median_ribo = median(percent_ribo), .groups = &quot;drop&quot;)

ggplot(metadata, aes(x = sample, y = percent_ribo, fill = group)) +
  geom_violin(scale = &quot;width&quot;, trim = F) +
  geom_point(data = medians, aes(x = sample, y = median_ribo), 
             color = &quot;black&quot;, size = 1, shape = 21, fill = &quot;black&quot;) + # Median dots
  scale_y_log10(labels = label_scientific()) + 
  scale_fill_manual(values = CFG$groupspal) +   
  theme_minimal() +                            
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size=20), axis.text.y = element_text(size=20), 
        legend.text = element_text(size = 20), legend.title = element_text(size=20),
        plot.title = element_text(size = 24, face = &quot;bold&quot;),
        panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank()) +
  labs(title = &quot;pct_ribo&quot;, x = &quot;&quot;, y = &quot;&quot;, fill = &quot;Group&quot;)</code></pre>
<p><img src="figure/Fig1_script.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dev.off()


# pdf(here(&quot;output&quot;, &quot;SupFigures&quot;,&quot;SupFig1B_3.pdf&quot;), width=14, height=8)
metadata$velocyto_unsplicedratio[metadata$velocyto_unsplicedratio == 0] &lt;- 0.1 # avoid inf values
medians &lt;- metadata %&gt;%
  group_by(sample) %&gt;%
  summarize(median_velo = median(velocyto_unsplicedratio), .groups = &quot;drop&quot;)

ggplot(metadata, aes(x = sample, y = velocyto_unsplicedratio, fill = group)) +
  geom_violin(scale = &quot;width&quot;, trim = F) +
  geom_point(data = medians, aes(x = sample, y = median_velo), 
             color = &quot;black&quot;, size = 1, shape = 21, fill = &quot;black&quot;) + # Median dots
  scale_y_log10(labels = label_scientific()) +  
  scale_fill_manual(values = CFG$groupspal) +  
  geom_hline(yintercept = 0.25) +
  theme_minimal() +                      
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size=20), axis.text.y = element_text(size=20), 
        legend.text = element_text(size = 20), legend.title = element_text(size=20),
        plot.title = element_text(size = 24, face = &quot;bold&quot;),
        panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank()) +
  labs(title = &quot;Intronic_Ratio&quot;, x = &quot;&quot;, y = &quot;&quot;, fill = &quot;Group&quot;)</code></pre>
<p><img src="figure/Fig1_script.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dev.off()</code></pre>
</div>
<div id="supp-fig1c" class="section level3">
<h3>Supp Fig1C</h3>
<pre class="r"><code>qc_thresholds &lt;- read.table(here(&quot;data&quot;,&quot;adult_QC_thresholds.txt&quot;), header = T, sep=&#39;\t&#39;)

samples&lt;-unique(adult_raw$sample)
samples &lt;- samples[order(factor(adult_raw$group[match(samples, adult_raw$sample)], levels = c(&quot;CTRL&quot;, &quot;MAD&quot;, &quot;RES&quot;, &quot;SAD&quot;)))]

# Apply functions to each sample
mito_vs_velo_plots &lt;- map(samples, function(sample_id) {
  print(paste(&quot;Processing sample:&quot;, sample_id))
  sObj &lt;- subset(adult_raw, subset = sample == sample_id)
  fun_mito_velo_plot(sObj, sample_id)
})</code></pre>
<pre><code>[1] &quot;Processing sample: S4R2&quot;
[1] &quot;Processing sample: S2R3&quot;
[1] &quot;Processing sample: S4R4&quot;
[1] &quot;Processing sample: S4R5&quot;
[1] &quot;Processing sample: S8R5&quot;
[1] &quot;Processing sample: S1R5stored&quot;
[1] &quot;Processing sample: S2R2&quot;
[1] &quot;Processing sample: S2R4&quot;
[1] &quot;Processing sample: S2R5&quot;
[1] &quot;Processing sample: S6R5&quot;
[1] &quot;Processing sample: S2R6&quot;
[1] &quot;Processing sample: S3R2&quot;
[1] &quot;Processing sample: S3R4&quot;
[1] &quot;Processing sample: S3R5&quot;
[1] &quot;Processing sample: S7R5&quot;
[1] &quot;Processing sample: S1R6&quot;
[1] &quot;Processing sample: S5R6&quot;
[1] &quot;Processing sample: S1R1&quot;
[1] &quot;Processing sample: S1R2&quot;
[1] &quot;Processing sample: S1R3&quot;
[1] &quot;Processing sample: S1R4&quot;
[1] &quot;Processing sample: S1R5&quot;
[1] &quot;Processing sample: S5R5&quot;
[1] &quot;Processing sample: S2R5stored&quot;</code></pre>
<pre class="r"><code># tiff(here(&quot;output&quot;, &quot;SupFigures&quot;,&quot;SupFig1C_1.tiff&quot;), width=24, height=12, units=&quot;in&quot;, res=300)
do.call(grid.arrange, c(mito_vs_velo_plots, nrow = 4, ncol = 6))</code></pre>
<pre><code>Warning: Removed 76 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 37 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 139 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 45 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 693 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 156 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 32 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 633 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 46 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 27 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 199 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 3 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 162 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 19 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 1927 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 385 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 2428 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 632 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 21 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 192 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 33 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 25 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 44 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<pre><code>Warning: Removed 84 rows containing non-finite outside the scale range
(`stat_bin2d()`).</code></pre>
<p><img src="figure/Fig1_script.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dev.off()

rm(adult_raw)
gc(full=T)</code></pre>
<pre><code>             used    (Mb)  gc trigger    (Mb) limit (Mb)   max used    (Mb)
Ncells   16452121   878.7    24413765  1303.9         NA   24413765  1303.9
Vcells 6263621347 47787.7 10241636446 78137.5     122880 7731082776 58983.5</code></pre>
</div>
</div>
<div id="raw-data-filtering-clustering" class="section level1">
<h1>Raw Data Filtering &amp; Clustering</h1>
<pre class="r"><code># adult_filtered &lt;- QC_filtering(adult_raw, qc_thresholds)
# 
# adult_filtered &lt;- NormalizeData(object = adult_filtered)
# adult_filtered &lt;- FindVariableFeatures(adult_filtered)
# adult_filtered &lt;- ScaleData(adult_filtered)
# adult_filtered &lt;- RunPCA(adult_filtered)
# 
# ElbowPlot(adult_filtered)
# 
# adult_filtered &lt;- FindNeighbors(adult_filtered, dims = 1:20)
# adult_filtered &lt;- FindClusters(adult_filtered, resolution = 1.4)
# adult_filtered &lt;- RunUMAP(adult_filtered, dims = 1:20, reduction = &quot;pca&quot;, reduction.name = &quot;umap.unintegrated&quot;)
#   
# adult_filtered &lt;- RunHarmony(adult_filtered, c(&quot;sample&quot;,&quot;Run&quot;), plot_convergence = TRUE,  kmeans_init_nstart=20, kmeans_init_iter_max=150)
# adult_filtered &lt;- RunUMAP(adult_filtered, reduction = &quot;harmony&quot;, dims = 1:20, reduction.name=&#39;harmony.umap&#39;) %&gt;%
#   FindNeighbors(reduction = &quot;harmony&quot;, dims = 1:20) %&gt;%
#   FindClusters(resolution = 1.4) %&gt;%
#   identity()

# saveRDS(adult_filtered, file=here(&quot;data&quot;, &quot;adult_filtered.RDS&quot;))</code></pre>
<pre class="r"><code># read filtered file
# adult&lt;-readRDS(here(&quot;data&quot;, &quot;adult_filtered.RDS&quot;))
adult&lt;-SetIdent(adult, value=&quot;broad_cell_type&quot;)</code></pre>
<div id="suppfig-1a---post-integration" class="section level3">
<h3>SuppFig 1A - post integration</h3>
<pre class="r"><code># tiff(here(&quot;output&quot;, &quot;SupFigures&quot;,&quot;SupFig1A_3.tiff&quot;), units=&quot;in&quot;, width=8, height=6, res=300)
DimPlot(adult, reduction = &quot;harmony.umap&quot;, group.by = &quot;group&quot;, label = F, cols = CFG$groupspal, pt.size = 0.01, raster=FALSE) + xlab(&quot;harmony.umap_1&quot;) + ylab(&quot;harmony.umap_2&quot;)+theme(axis.text.x=element_text(size=20),axis.text.y=element_text(size=20), axis.title.x = element_text(size=20), axis.title.y = element_text(size=20), legend.text = element_text(size=20), title = element_text(size=20))</code></pre>
<p><img src="figure/Fig1_script.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dev.off()


# tiff(here(&quot;output&quot;, &quot;SupFigures&quot;,&quot;SupFig1A_4.tiff&quot;), units=&quot;in&quot;, width=8, height=6, res=300)
DimPlot(adult, reduction = &quot;harmony.umap&quot;, group.by = &quot;Run&quot;, label = F, cols = CFG$mypal[11:16], pt.size = 0.01, raster=FALSE) + xlab(&quot;harmony.umap_1&quot;) + ylab(&quot;harmony.umap_2&quot;)+theme(axis.text.x=element_text(size=20),axis.text.y=element_text(size=20), axis.title.x = element_text(size=20), axis.title.y = element_text(size=20), legend.text = element_text(size=20), title = element_text(size=20))</code></pre>
<p><img src="figure/Fig1_script.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dev.off()

# tiff(here(&quot;output&quot;, &quot;SupFigures&quot;,&quot;SupFig1A_5.tiff&quot;), units=&quot;in&quot;, width=8, height=6, res=300)
DimPlot(adult, reduction = &quot;harmony.umap&quot;, group.by = &quot;sample&quot;, label = F, cols = CFG$mypal, pt.size = 0.01, raster=FALSE) + xlab(&quot;harmony.umap_1&quot;) + ylab(&quot;harmony.umap_2&quot;)+theme(axis.text.x=element_text(size=20),axis.text.y=element_text(size=20), axis.title.x = element_text(size=20), axis.title.y = element_text(size=20), legend.text = element_text(size=20), title = element_text(size=20))</code></pre>
<p><img src="figure/Fig1_script.Rmd/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dev.off()</code></pre>
</div>
<div id="fig-1a---prepared-manually" class="section level3">
<h3>Fig 1A - Prepared manually</h3>
</div>
<div id="fig-1b" class="section level3">
<h3>Fig 1B</h3>
<pre class="r"><code>Donor_info_heatmap &lt;- as.data.frame(read_excel(here(&quot;data&quot;, &quot;Donor_info_heatmap.xlsx&quot;)))

# === PREPARE DATA ===
## Define character columns
chr_cols &lt;- c(&quot;Sample_name&quot;, &quot;Group&quot;, &quot;Sex&quot;, &quot;APOE&quot;)

## Extract numeric columns and convert to numeric
data_numeric &lt;- Donor_info_heatmap %&gt;% dplyr::select(-all_of(chr_cols)) %&gt;% mutate(across(everything(), as.numeric))</code></pre>
<pre><code>Warning: There were 5 warnings in `mutate()`.
The first warning was:
ℹ In argument: `across(everything(), as.numeric)`.
Caused by warning:
! NAs introduced by coercion
ℹ Run `dplyr::last_dplyr_warnings()` to see the 4 remaining warnings.</code></pre>
<pre class="r"><code>## Combine numeric and character columns
data &lt;- Donor_info_heatmap %&gt;% dplyr::select(all_of(chr_cols)) %&gt;% bind_cols(data_numeric)

# === DEFINE COLORS ===
## Color gradients for numeric columns
numeric_columns &lt;- names(data_numeric)
color_gradients &lt;- lapply(numeric_columns, function(col_name) {
  min_val &lt;- min(na.omit(data_numeric[[col_name]]))
  max_val &lt;- max(na.omit(data_numeric[[col_name]]))
  circlize::colorRamp2(
    breaks = c(min_val, max_val),
    colors = c(&quot;lightgrey&quot;, &quot;#542437&quot;)
  )
})
names(color_gradients) &lt;- numeric_columns

## Categorical colors
categorical_cols &lt;- list(
  Group = c(&quot;Control&quot; = &quot;#A3BF8C&quot;, &quot;Moderate_AD&quot; = &quot;#D08770&quot;, &quot;Resilient&quot;= &quot;#808EAF&quot;, &quot;Severe_AD&quot;= &quot;#813E3C&quot;),
  Sex = c(&quot;F&quot;=&quot;#A5C2B2&quot;, &quot;M&quot;= &quot;#6C947F&quot;),
  APOE = c(&quot;33&quot;=&quot;#A3B6CB&quot;,  &quot;43&quot;=&quot;#548FA7&quot;, &quot;44&quot;=&quot;#59707E&quot;, &quot;NA&quot;= &quot;ivory2&quot;)
)

## Merge colors
heatmap_colors &lt;- c(color_gradients, categorical_cols)

# === CREATE HEATMAP ===
## Annotation
ha &lt;- HeatmapAnnotation(df=data %&gt;% dplyr::select(-Sample_name), col = heatmap_colors, na_col = &quot;ivory2&quot;)

## Dummy matrix for heatmap
dummy_matrix &lt;- matrix(1, nrow = 0, ncol = 24)

## Heatmap object
heatmap &lt;- Heatmap(dummy_matrix, 
                   name = &quot;Clinical Data&quot;, 
                   top_annotation = ha, 
                   cluster_columns = FALSE, 
                   cluster_rows = FALSE,
                   show_column_names = TRUE,
                   show_row_names = TRUE,
                   na_col = &quot;ivory2&quot;,  # Color for NA values
                   heatmap_legend_param = list(
                     title = &quot;Heatmap Legend&quot;,
                     legend_direction = &quot;horizontal&quot;
                   ))

# === EXPORT HEATMAP ===
# pdf(here(&quot;output&quot;, &quot;Figures&quot;, &quot;Figure1B.pdf&quot;), height = 8, width = 10)
draw(heatmap, annotation_legend_side = &quot;bottom&quot;)</code></pre>
<p><img src="figure/Fig1_script.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dev.off()</code></pre>
</div>
<div id="fig-1c" class="section level3">
<h3>Fig 1C</h3>
<pre class="r"><code># === UPPER PANEL UMAP ===
# tiff(here(&quot;output&quot;, &quot;Figures&quot;,&quot;Figure1C_1.tiff&quot;), units=&quot;in&quot;, width=8, height=6, res=300)
DimPlot(adult, reduction = &quot;harmony.umap&quot;, group.by = &quot;broad_cell_type&quot;, label = F, cols = c(&quot;#F28C8F&quot;, &quot;#ECCF76&quot;,  &quot;#6ABEAE&quot;,  &quot;#C3894C&quot;, &quot;#147F84&quot;, &quot;#C69D9C&quot;, &quot;#E86562&quot;, &quot;#B48DAC&quot;, &quot;#F2B8A5&quot;), pt.size = 0.5, raster=FALSE) + xlab(&quot;harmony.umap_1&quot;) + ylab(&quot;harmony.umap_2&quot;)</code></pre>
<p><img src="figure/Fig1_script.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dev.off()

# === CREATE THE FEATUREPLOTS ===
features &lt;- c(&quot;AQP4&quot;, &quot;CLDN5&quot;, &quot;CFAP299&quot;, &quot;PROX1&quot;, &quot;RBFOX3&quot;, &quot;GAD2&quot;, &quot;PTPRC&quot;, &quot;MOG&quot;, &quot;PDGFRA&quot;, &quot;DCN&quot;)
plots &lt;- lapply(features, create_feature_plot, data = adult, gradient_colors = CFG$gradient_white, label=F, raster = F, order=T)</code></pre>
<pre><code>Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.
Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.
Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.
Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.
Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.
Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.
Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.
Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.
Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.
Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.</code></pre>
<pre class="r"><code># tiff(here(&quot;output&quot;, &quot;Figures&quot;, &quot;Figure1C_2.tiff&quot;), units = &quot;in&quot;, width = 40, height = 12, res = 300)
ggarrange(plotlist = plots, ncol = 5, nrow = 2)</code></pre>
<p><img src="figure/Fig1_script.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dev.off()</code></pre>
</div>
<div id="fig-1d" class="section level3">
<h3>Fig 1D</h3>
<pre class="r"><code>genes&lt;- c(&quot;RBFOX3&quot;, &quot;SLC17A7&quot;,  &quot;CFAP299&quot;, &quot;SV2B&quot;, &quot;PROX1&quot;, &quot;SEMA5A&quot;,  &quot;COL25A1&quot;, &quot;GAD2&quot;, &quot;GAD1&quot;, &quot;PDGFRA&quot;, &quot;NEU4&quot;, &quot;OLIG1&quot;, &quot;MOG&quot;, &quot;MAG&quot;, &quot;MOBP&quot;, &quot;PTPRC&quot;, &quot;ITGAM&quot;, &quot;CSF1R&quot;,  &quot;AQP4&quot;, &quot;GJA1&quot;, &quot;ALDOC&quot;, &quot;S100B&quot;, &quot;CLDN5&quot;, &quot;CD34&quot;, &quot;VWF&quot;, &quot;DCN&quot;, &quot;COL1A2&quot;)
myLevels&lt;-c(&quot;ExN_MC/CA&quot;, &quot;ExN_GC&quot;, &quot;InN&quot;, &quot;OPC&quot;, &quot;Oligodendrocyte&quot;, &quot;Microglia&quot;, &quot;Astrocyte&quot;, &quot;Endothelia&quot;, &quot;VLMC&quot;)      
Idents(adult) &lt;- factor(Idents(adult), levels= rev(myLevels))

# pdf(here(&quot;output&quot;, &quot;Figures&quot;, &quot;Figure1D.pdf&quot;), width=12, height=6)
DotPlot(adult, features = genes, cols=CFG$gradient_white) +theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size=18), axis.text.y = element_text(size=18), legend.text = element_text(size = 18))</code></pre>
<p><img src="figure/Fig1_script.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dev.off()</code></pre>
</div>
<div id="fig-1e" class="section level3">
<h3>Fig 1E</h3>
<pre class="r"><code># === UMAP SPLIT BY GROUP ===
colors&lt;-c(&quot;#F28C8F&quot;, &quot;#ECCF76&quot;,  &quot;#6ABEAE&quot;,  &quot;#C3894C&quot;, &quot;#147F84&quot;, &quot;#C69D9C&quot;, &quot;#E86562&quot;, &quot;#B48DAC&quot;, &quot;#F2B8A5&quot;)

# tiff(here(&quot;output&quot;, &quot;Figures&quot;,&quot;Figure1E_1.tiff&quot;), units=&quot;in&quot;,width=24, height=6, res=300)
DimPlot(adult, reduction = &quot;harmony.umap&quot;, split.by = &quot;group&quot;, group.by = &quot;broad_cell_type&quot;, label = F, cols = colors, pt.size = 0.5, raster=FALSE) + xlab(&quot;harmony.umap_1&quot;) + ylab(&quot;harmony.umap_2&quot;)</code></pre>
<p><img src="figure/Fig1_script.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dev.off()

# === PROPELLER BAR PLOT ===
output.logit &lt;- propeller(clusters=adult$broad_cell_type, sample=adult$sample, group=adult$group, transform=&quot;logit&quot;)</code></pre>
<pre><code>Performing logit transformation of proportions</code></pre>
<pre><code>group variable has &gt; 2 levels, ANOVA will be performed</code></pre>
<pre class="r"><code>props &lt;- getTransformedProps(clusters=adult$broad_cell_type, sample=adult$group, transform=&quot;logit&quot;)</code></pre>
<pre><code>Performing logit transformation of proportions</code></pre>
<pre class="r"><code>props_df&lt;-as.data.frame(props$Proportions)

# pdf(here(&quot;output&quot;, &quot;Figures&quot;,&quot;Figure1E_2.pdf&quot;), width=8, height=6)
ggplot(props_df, aes(fill=clusters, y=Freq, x=sample)) + scale_fill_manual(values=colors) + geom_bar(position=&quot;stack&quot;, stat=&quot;identity&quot;) + theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),panel.grid.minor = element_blank(), axis.text.x = element_text(size=14), axis.text.y = element_text(size=14), legend.text = element_text(size = 14), legend.title = element_text(size = 14))</code></pre>
<p><img src="figure/Fig1_script.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dev.off()</code></pre>
</div>
<div id="supfig-1d" class="section level3">
<h3>SupFig 1D</h3>
<pre class="r"><code>data &lt;- data.frame(
  Study = c(&quot;Yao&quot;, &quot;Habib&quot;, &quot;Ayhan&quot;, &quot;Franjic&quot;, &quot;Zhou&quot;, &quot;This Study&quot;),
  GC_Percentage = c(19, 10, 7, 23, 23, 53),
  Non_GC_Percentage = 100 - c(19, 10, 7, 23, 23, 53)
)

data_long &lt;- melt(data, id.vars = &quot;Study&quot;, 
                  variable.name = &quot;Cell_Type&quot;, 
                  value.name = &quot;Percentage&quot;)</code></pre>
<pre><code>Warning in melt(data, id.vars = &quot;Study&quot;, variable.name = &quot;Cell_Type&quot;,
value.name = &quot;Percentage&quot;): The melt generic in data.table has been passed a
data.frame and will attempt to redirect to the relevant reshape2 method; please
note that reshape2 is deprecated, and this redirection is now deprecated as
well. To continue using melt methods from reshape2 while both libraries are
attached, e.g. melt.list, you can prepend the namespace like
reshape2::melt(data). In the next version, this warning will become an error.</code></pre>
<pre class="r"><code># Sort the Study factor levels based on GC_Percentage
data_long$Study &lt;- factor(data_long$Study, levels = data$Study[order(data$GC_Percentage)])

# Reorder the levels of Cell_Type to ensure GC_Percentage is stacked first
data_long$Cell_Type &lt;- factor(data_long$Cell_Type, levels = c(&quot;Non_GC_Percentage&quot;, &quot;GC_Percentage&quot;))


# tiff(here(&quot;output&quot;, &quot;SupFigures&quot;, &quot;SupFig1D.tiff&quot;), width = 5, height = 10, units = &quot;in&quot;, res = 300)
ggplot(data_long, aes(x = Study, y = Percentage, fill = Cell_Type)) +
  geom_bar(stat = &quot;identity&quot;, width = 0.7) +
  scale_fill_manual(values = c(&quot;GC_Percentage&quot; = &quot;#A3B6CB&quot;, &quot;Non_GC_Percentage&quot; = &quot;lightgrey&quot;)) +
  labs(title = &quot;Percentage of Granule Cells in Various Studies&quot;,
       x = &quot;Study&quot;,
       y = &quot;Percentage of Cells&quot;,
       fill = &quot;Cell Type&quot;) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        panel.grid = element_blank()) </code></pre>
<p><img src="figure/Fig1_script.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># dev.off()</code></pre>
</div>
<div id="suptable1.1" class="section level3">
<h3>SupTable1.1</h3>
<pre class="r"><code># Table with number of cells per cell type in donors and groups
# table&lt;-as.data.frame(table(adult@meta.data$broad_cell_type, adult@meta.data$sample))
# 
# data_table &lt;- table %&gt;%
#   pivot_wider(names_from = Var2, values_from = Freq) %&gt;%
#   column_to_rownames(var = &quot;Var1&quot;)
# 
# head(table)

# write.table(data_table, here(&quot;output&quot;, &quot;SupTables&quot;,&quot;SupTable1_1.txt&quot;), quote = F, sep=&quot;\t&quot;, row.names = T)</code></pre>
</div>
<div id="suptable1.2" class="section level3">
<h3>SupTable1.2</h3>
<pre class="r"><code># Convert Seurat object to pseudo-bulk counts
# pb &lt;- edgeR::Seurat2PB(adult, sample = &quot;sample&quot;, cluster = &quot;broad_cell_type&quot;)
# counts &lt;- pb$counts
# 
# # Prepare metadata
# metadata &lt;- data.frame(cluster = gsub(&quot;cluster&quot;, &quot;&quot;, unlist(str_split_fixed(colnames(pb), &quot;_&quot;, 2))[, 2]))
# rownames(metadata) &lt;- colnames(pb)
# 
# # Apply deseq2 function to each cluster
# clusters &lt;- unique(adult$broad_cell_type)
# DE_results &lt;- fun_deseq2_cellType(counts, metadata,
#                                            smallestGroupSize = 20, 
#                                            use_vst = TRUE, 
#                                            padj_threshold = 0.05, 
#                                            log2FC_threshold = 0.5, 
#                                            minExpression = 10)
# 
# # Combine all results
# DE_result_list &lt;- bind_rows(DE_result_list)
# 
# head(DE_result_list)
# write.table(DE_result_list, here(&quot;output&quot;, &quot;SupTables&quot;,&quot;SupTable1_2.txt&quot;), sep=&quot;\t&quot;, quote = F, row.names = F)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

Random number generation:
 RNG:     Mersenne-Twister 
 Normal:  Inversion 
 Sample:  Rounding 
 
locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Amsterdam
tzcode source: internal

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] viridis_0.6.4               viridisLite_0.4.2          
 [3] UCell_2.6.2                 tradeSeq_1.16.0            
 [5] lubridate_1.9.3             forcats_1.0.0              
 [7] readr_2.1.5                 tidyverse_2.0.0            
 [9] yardstick_1.2.0             workflowsets_1.0.1         
[11] workflows_1.1.3             tune_1.1.2                 
[13] tidyr_1.3.0                 rsample_1.2.0              
[15] recipes_1.0.8               parsnip_1.1.1              
[17] modeldata_1.2.0             infer_1.0.5                
[19] dials_1.2.0                 scales_1.3.0               
[21] broom_1.0.5                 tidymodels_1.1.1           
[23] tibble_3.2.1                stringr_1.5.1              
[25] speckle_1.2.0               slingshot_2.10.0           
[27] TrajectoryUtils_1.10.0      princurve_2.1.6            
[29] SCP_0.5.6                   scDblFinder_1.16.0         
[31] scCustomize_1.1.3           SeuratObject_5.0.1         
[33] Seurat_4.4.0                readxl_1.4.3               
[35] RColorBrewer_1.1-3          purrr_1.0.2                
[37] pheatmap_1.0.12             org.Hs.eg.db_3.18.0        
[39] AnnotationDbi_1.64.1        Nebulosa_1.12.0            
[41] patchwork_1.3.0.9000        nebula_1.5.2               
[43] monocle3_1.3.4              Matrix.utils_0.9.7         
[45] Matrix_1.6-4                magrittr_2.0.3             
[47] ktplots_2.3.0               harmony_1.1.0              
[49] Rcpp_1.0.14                 gridExtra_2.3              
[51] ggsignif_0.6.4              ggpubr_0.6.0               
[53] ggnewscale_0.4.9            ggbeeswarm_0.7.2           
[55] enrichplot_1.22.0           EnhancedVolcano_1.20.0     
[57] ggrepel_0.9.4               edgeR_4.0.2                
[59] limma_3.58.1                easyGgplot2_1.0.0.9000     
[61] dunn.test_1.3.6             DropletUtils_1.22.0        
[63] SingleCellExperiment_1.24.0 DESeq2_1.42.0              
[65] SummarizedExperiment_1.32.0 MatrixGenerics_1.14.0      
[67] matrixStats_1.1.0           GenomicRanges_1.54.1       
[69] GenomeInfoDb_1.38.1         IRanges_2.36.0             
[71] S4Vectors_0.40.2            data.table_1.14.8          
[73] cowplot_1.1.1               ComplexHeatmap_2.21.2      
[75] clusterProfiler_4.10.1      cli_3.6.4                  
[77] CellChat_2.1.2              Biobase_2.62.0             
[79] BiocGenerics_0.48.1         ggplot2_3.5.1              
[81] igraph_1.5.1                dplyr_1.1.4                
[83] biomaRt_2.58.0              BiocParallel_1.36.0        
[85] here_1.0.1                  workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] ica_1.0-3                 plotly_4.10.4            
  [3] scater_1.30.0             rematch2_2.1.2           
  [5] devtools_2.4.5            zlibbioc_1.48.0          
  [7] GPfit_1.0-8               tidyselect_1.2.0         
  [9] bit_4.0.5                 doParallel_1.0.17        
 [11] clue_0.3-65               lattice_0.22-5           
 [13] rjson_0.2.21              blob_1.2.4               
 [15] urlchecker_1.0.1          rngtools_1.5.2           
 [17] S4Arrays_1.2.0            parallel_4.3.1           
 [19] png_0.1-8                 ggplotify_0.1.2          
 [21] registry_0.5-1            goftest_1.2-3            
 [23] BiocIO_1.12.0             bluster_1.12.0           
 [25] grr_0.9.5                 lhs_1.1.6                
 [27] Signac_1.12.0             BiocNeighbors_1.20.0     
 [29] ggnetwork_0.5.13          uwot_0.1.16              
 [31] shadowtext_0.1.2          curl_5.1.0               
 [33] mime_0.12                 evaluate_0.23            
 [35] tidytree_0.4.5            leiden_0.4.3.1           
 [37] stringi_1.8.2             backports_1.4.1          
 [39] XML_3.99-0.16             httpuv_1.6.12            
 [41] paletteer_1.5.0           rappdirs_0.3.3           
 [43] splines_4.3.1             RcppRoll_0.3.0           
 [45] mclust_6.0.1              prodlim_2023.08.28       
 [47] doRNG_1.8.6               ggraph_2.1.0             
 [49] sctransform_0.4.1         sessioninfo_1.2.2        
 [51] DBI_1.1.3                 terra_1.7-55             
 [53] HDF5Array_1.30.0          jquerylib_0.1.4          
 [55] withr_3.0.2               git2r_0.35.0             
 [57] class_7.3-22              systemfonts_1.0.5        
 [59] xgboost_1.7.6.1           rprojroot_2.0.4          
 [61] lmtest_0.9-40             tidygraph_1.2.3          
 [63] rtracklayer_1.62.0        BiocManager_1.30.22      
 [65] htmlwidgets_1.6.4         fs_1.6.3                 
 [67] trust_0.1-8               statnet.common_4.10.0    
 [69] labeling_0.4.3            SparseArray_1.2.2        
 [71] cellranger_1.1.0          RcppZiggurat_0.1.6       
 [73] reticulate_1.34.0         zoo_1.8-12               
 [75] XVector_0.42.0            knitr_1.45               
 [77] network_1.18.2            timechange_0.2.0         
 [79] foreach_1.5.2             timeDate_4022.108        
 [81] ggtree_3.10.0             rhdf5_2.46.1             
 [83] R.oo_1.25.0               RSpectra_0.16-1          
 [85] irlba_2.3.5.1             ggrastr_1.0.2            
 [87] gridGraphics_0.5-1        ellipsis_0.3.2           
 [89] lazyeval_0.2.2            yaml_2.3.7               
 [91] survival_3.5-7            scattermore_1.2          
 [93] crayon_1.5.2              RcppAnnoy_0.0.21         
 [95] progressr_0.14.0          tweenr_2.0.2             
 [97] later_1.3.1               ggridges_0.5.4           
 [99] codetools_0.2-19          GlobalOptions_0.1.2      
[101] profvis_0.3.8             KEGGREST_1.42.0          
[103] Rtsne_0.16                shape_1.4.6              
[105] Rsamtools_2.18.0          filelock_1.0.2           
[107] DiceDesign_1.9            pkgconfig_2.0.3          
[109] xml2_1.3.6                GenomicAlignments_1.38.0 
[111] getPass_0.2-4             aplot_0.2.2              
[113] spatstat.sparse_3.0-3     ape_5.7-1                
[115] gridBase_0.4-7            xtable_1.8-4             
[117] highr_0.10                car_3.1-2                
[119] plyr_1.8.9                httr_1.4.7               
[121] tools_4.3.1               globals_0.16.2           
[123] hardhat_1.3.0             pkgbuild_1.4.2           
[125] Rfast_2.1.0               beeswarm_0.4.0           
[127] nlme_3.1-164              HDO.db_0.99.1            
[129] dbplyr_2.4.0              lme4_1.1-35.1            
[131] digest_0.6.33             tzdb_0.4.0               
[133] furrr_0.3.1               farver_2.1.2             
[135] reshape2_1.4.4            ks_1.14.1                
[137] yulab.utils_0.1.0         rpart_4.1.23             
[139] glue_1.8.0                cachem_1.0.8             
[141] BiocFileCache_2.10.1      polyclip_1.10-6          
[143] proxyC_0.3.4              generics_0.1.3           
[145] Biostrings_2.70.1         ggalluvial_0.12.5        
[147] mvtnorm_1.2-4             parallelly_1.36.0        
[149] pkgload_1.3.3             statmod_1.5.0            
[151] R.cache_0.16.0            ScaledMatrix_1.10.0      
[153] carData_3.0-5             minqa_1.2.6              
[155] pbapply_1.7-2             spam_2.10-0              
[157] gson_0.1.0                dqrng_0.3.2              
[159] gower_1.0.1               graphlayouts_1.0.2       
[161] gtools_3.9.5              lava_1.7.3               
[163] shiny_1.8.0               GenomeInfoDbData_1.2.11  
[165] R.utils_2.12.3            rhdf5filters_1.14.1      
[167] RCurl_1.98-1.13           memoise_2.0.1            
[169] rmarkdown_2.25            R.methodsS3_1.8.2        
[171] future_1.33.0             doFuture_1.0.1           
[173] svglite_2.1.3             RANN_2.6.1               
[175] Cairo_1.6-2               spatstat.data_3.0-3      
[177] rstudioapi_0.15.0         janitor_2.2.0            
[179] cluster_2.1.6             whisker_0.4.1            
[181] spatstat.utils_3.1-3      hms_1.1.3                
[183] fitdistrplus_1.1-11       munsell_0.5.1            
[185] colorspace_2.1-1          FNN_1.1.3.2              
[187] rlang_1.1.5               DelayedMatrixStats_1.24.0
[189] sparseMatrixStats_1.14.0  ipred_0.9-14             
[191] dotCall64_1.1-1           ggforce_0.4.1            
[193] circlize_0.4.16           scuttle_1.12.0           
[195] mgcv_1.9-0                xfun_0.52                
[197] coda_0.19-4               sna_2.8                  
[199] remotes_2.4.2.1           iterators_1.0.14         
[201] abind_1.4-5               GOSemSim_2.28.0          
[203] treeio_1.26.0             Rhdf5lib_1.24.0          
[205] bitops_1.0-7              ps_1.7.5                 
[207] promises_1.2.1            scatterpie_0.2.1         
[209] RSQLite_2.3.3             qvalue_2.34.0            
[211] fgsea_1.28.0              DelayedArray_0.28.0      
[213] GO.db_3.18.0              compiler_4.3.1           
[215] prettyunits_1.2.0         boot_1.3-28.1            
[217] beachmat_2.18.0           listenv_0.9.0            
[219] BiocSingular_1.18.0       tensor_1.5               
[221] usethis_2.2.2             MASS_7.3-60              
[223] progress_1.2.2            spatstat.random_3.2-2    
[225] R6_2.6.1                  fastmap_1.1.1            
[227] fastmatch_1.1-4           rstatix_0.7.2            
[229] vipor_0.4.5               ROCR_1.0-11              
[231] nnet_7.3-19               rsvd_1.0.5               
[233] gtable_0.3.6              KernSmooth_2.23-22       
[235] miniUI_0.1.1.1            deldir_2.0-2             
[237] htmltools_0.5.7           RcppParallel_5.1.7       
[239] bit64_4.0.5               ggprism_1.0.4            
[241] spatstat.explore_3.2-5    lifecycle_1.0.4          
[243] processx_3.8.2            restfulr_0.0.15          
[245] nloptr_2.0.3              callr_3.7.3              
[247] sass_0.4.7                vctrs_0.6.5              
[249] snakecase_0.11.1          spatstat.geom_3.2-7      
[251] DOSE_3.28.1               scran_1.30.0             
[253] NMF_0.28                  ggfun_0.1.3              
[255] sp_2.1-2                  future.apply_1.11.0      
[257] pracma_2.4.4              bslib_0.6.1              
[259] pillar_1.10.2             magick_2.8.5             
[261] metapod_1.10.0            locfit_1.5-9.8           
[263] jsonlite_1.8.8            GetoptLong_1.0.5         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
